src/runtime/virtcontainers/hypervisor_linux_amd64.go

```diff
@@ -17,17 +19,34 @@ const (
 	snpKvmParameterPath = "/sys/module/kvm_amd/parameters/sev_snp"
 )
 
+// TDX is supported and properly loaded when the firmware directory (either tdx or tdx_seam) exists or `tdx` is part of the CPU flag
+func checkTdxGuestProtection(flags map[string]bool) bool {
+	if flags[tdxCPUFlag] {
+		return true
+	}
+
+	if d, err := os.Stat(tdxSysFirmwareDir); err == nil && d.IsDir() {
```

Comment:
Maybe you should treat the error better. The directory might exist but you got no permission to stat and returning that tdx guest protection is false is misleading.

Reply:
As `containerd-shim-v2` runs as `root`, I sincerely think we're fine on how we're dealing with this.